src/Sylius/Component/Resource/Model/TranslatableTrait.php

```diff
@@ -78,11 +78,13 @@ public function getTranslation($locale = null)
             return $translation;
         }
 
-        $fallbackTranslation = $this->translations->get($this->fallbackLocale);
-        if (null !== $fallbackTranslation) {
-            $this->translationsCache[$this->fallbackLocale] = $fallbackTranslation;
+        if ($locale !== $this->fallbackLocale) {
+            $fallbackTranslation = $this->translations->get($this->fallbackLocale);
```

Comment:
Yes, IIRC `$this->translations->get(...)` almost always results in a query, cache was introduced to solve it. :)